src/smefit/optimize/ns.py

```diff
@@ -105,6 +107,11 @@ def from_dict(cls, config):
             )
         coefficients = CoefficientManager.from_dict(config["coefficients"])
 
+        if "single_parameter_fits" not in config:
```

Comment:
here and in MC you can use:

```suggestion
       single_parameter_fits = config.get("single_parameter_fits", False)
```